awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsMetricAttributeGenerator.java

```diff
@@ -455,6 +456,8 @@ private static void setRemoteResourceTypeAndIdentifier(SpanData span, Attributes
         remoteResourceType = Optional.of(NORMALIZED_BEDROCK_SERVICE_NAME + "::Guardrail");
         remoteResourceIdentifier =
             Optional.ofNullable(escapeDelimiters(span.getAttributes().get(AWS_GUARDRAIL_ID)));
+        cloudformationPrimaryIdentifier =
+            Optional.ofNullable(escapeDelimiters(span.getAttributes().get(AWS_GUARDRAIL_ARN)));
```

Comment:
I think in upstream[ PR conversation](https://github.com/mxiamxia/opentelemetry-java-instrumentation/pull/13#discussion_r1795942737), we want to "extract remote_resource_identifier from cloudformation_primary_identifier to make sure the consistency of two values."? So the `AWS_GUARDRAIL_ID` won't be directly used here?

Reply:
I think as long as the relationship between the `remote_resource_identifier` and `cloudformation_primary_identifier` is consistent with the pattern

```
remote_resource_identifier = <some_identifier>
cloudformation_primary_identifier = <arn_prefix>:<some_identifier>
```

it should be okay to directly assign these attributes instead of extracting if they are easily accessible from the request/response bodies. I think introducing logic to manually parse strings may increase risk of bugs in the future.